### PR TITLE
feat: twitter langchain

### PR DESCRIPTION
**Pull request by stat**
### What changed? Why?

- Adding support for Twitter actions

### CODE REVIEW

File: cdp-agentkit-core/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,10 @@
 
 - Improved prompts for all actions.
 
+## [0.0.6] - 2024-12-03
+
+### Fixed
+
```
Comment (John-peterson-coinbase) [inline cdp-agentkit-core/CHANGELOG.md:14]: version should be `0.0.8` for this release. Please add a description of what social features are added

File: twitter-langchain/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  verbose: true,
+});
+
+// Example usage
+const result = await executor.invoke({
+  input: "please post 'hello, world!' to twitter"
+});
+
+console.log(result.output);
+```
```
Comment (John-peterson-coinbase) [inline twitter-langchain/README.md:80]: Please update to use `createReactAgent` pattern like https://github.com/coinbase/cdp-agentkit-nodejs/blob/master/cdp-langchain/README.md

File: twitter-langchain/examples/chatbot/.env-local [hunk 3]
```diff
@@ -0,0 +1,5 @@
+OPENAI_API_KEY=
+TWITTER_ACCESS_TOKEN=
+TWITTER_ACCESS_TOKEN_SECRET=
+TWITTER_API_KEY=
```
Comment (John-peterson-coinbase) [inline twitter-langchain/examples/chatbot/.env-local:4]: the bearer token is also needed?

File: twitter-langchain/examples/chatbot/chatbot.ts [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+async function initialize() {
+  // Initialize LLM
+  const llm = new ChatOpenAI({ model: "gpt-4o-mini" });
+
+  // Twitter (X) Agentkit options - these are loaded automatically via the ENV
+  const options = {
+    apiKey: "$TWITTER_API_KEY",
+    apiSecret: "$TWITTER_API_SECRET",
+    accessToken: "$TWITTER_ACCESS_TOKEN",
+    accessTokenSecret: "$TWITTER_ACCESS_TOKEN_SECRET",
```
Comment (John-peterson-coinbase) [inline twitter-langchain/examples/chatbot/chatbot.ts:37]: Bearer token is also required?

**Review by John-peterson-coinbase — COMMENTED** [review]
A few small comments. Otherwise LGTM!

File: twitter-langchain/package.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    "ai",
+    "agent",
+    "nodejs",
+    "typescript",
+    "twitter",
+    "langchain"
+  ],
+  "dependencies": {
+    "@coinbase/cdp-agentkit-core": "^0.0.8",
+    "@langchain/core": "0.3.21",
```
Comment (derek-cb) [inline twitter-langchain/package.json:38]: this should probably match what we have in cdp-agentkit-core package.json (same for Twitter chatbot). perhaps we should update the version in core as well?

Comment (derek-cb) [reply twitter-langchain/package.json:38]: or alternatively, we can downgrade this to match core, so as to reduce any risk of destabilizing core at the moment

File: cdp-agentkit-core/CHANGELOG.md (hunk 1 above)
Comment (stat) [reply cdp-agentkit-core/CHANGELOG.md:14]: roger that!

File: twitter-langchain/README.md (hunk 2 above)
Comment (stat) [reply twitter-langchain/README.md:80]: roger that

File: twitter-langchain/examples/chatbot/chatbot.ts (hunk 4 above)
Comment (stat) [reply twitter-langchain/examples/chatbot/chatbot.ts:37]: <c/p from the prev comment>

this is an oddity that i dug into. there is a discrepancy between the libs being used in agentkit.py and agentkit.js. the python twitter library will not get account mentions without the bearer token provided however, the nodejs lib is happy with just the app and token keys and secrets. 

the bearer token is used for twitter-api-v2 integration. i think the nodejs libary is either calling the v1 mentions, or is obtaining a v2 token behind the scenes.

was thinking that maybe we could try another twitter library for agentkit.py, or maybe patch twit to remove the bearer token dep?

File: twitter-langchain/examples/chatbot/.env-local (hunk 3 above)
Comment (stat) [reply twitter-langchain/examples/chatbot/.env-local:4]: this is an oddity that i dug into. there is a discrepancy between the libs being used in agentkit.py and agentkit.js. the python twitter library will not get account mentions without the bearer token provided however, the nodejs lib is happy with just the app and token keys and secrets. 

the bearer token is used for twitter-api-v2 integration. i think the nodejs libary is either calling the v1 mentions, or is obtaining a v2 token behind the scenes.

was thinking that maybe we could try another twitter library for agentkit.py, or maybe patch twit to remove the bearer token dep?

File: twitter-langchain/examples/chatbot/chatbot.ts [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+  };
+
+  // Twitter (X) Agentkit
+  const twitterAgentkit = new TwitterAgentkit(/*options*/);
```
Comment (derek-cb) [inline twitter-langchain/examples/chatbot/chatbot.ts:41]: curious why the options param is commented out?

Comment (stat) [reply twitter-langchain/examples/chatbot/chatbot.ts:41]: just wanted to show it's an optional param if the user wants to override the ENV load for Twitter (X) app and token keys/secrets

File: twitter-langchain/package.json (hunk 5 above)
Comment (stat) [reply twitter-langchain/package.json:38]: yeah, so core was bumped to 0.0.8, and both twitter-langchain and the twitter chatbot use 0.0.8 and langchain/core 0.3.21

Comment (stat) [reply twitter-langchain/package.json:38]: do you want me to bump to 0.0.9?

File: twitter-langchain/examples/chatbot/chatbot.ts (hunk 6 above)
Comment (stat) [reply twitter-langchain/examples/chatbot/chatbot.ts:42]: added a comment that should help clarify!

File: twitter-langchain/package.json (hunk 5 above)
Comment (derek-cb) [reply twitter-langchain/package.json:38]: UPDATE: meant to say that the `@langchain/core` package version should probably match what we have in `cdp-langchain`'s package.json (same for Twitter chatbot). perhaps we should update the version in `cdp-langchain` as well, or downgrade the version here to ensure feature parity between the two.

Comment (derek-cb) [reply twitter-langchain/package.json:38]: (small nit)

File: twitter-langchain/examples/chatbot/chatbot.ts (hunk 6 above)
Comment (derek-cb) [reply twitter-langchain/examples/chatbot/chatbot.ts:42]: thanks @stat. it's a nit but honestly I'd prefer removing the options var entirely in the Twitter `chatbot.ts` - like we do in the `cdp-langchain` chatbot when initializing `CdpAgentkit` - so as to discourage its usage and any accidents in people committing secrets in their code

Comment (stat) [reply twitter-langchain/examples/chatbot/chatbot.ts:42]: kk, roger that

File: twitter-langchain/package.json (hunk 5 above)
Comment (stat) [reply twitter-langchain/package.json:38]: ahh roger that

**Review by derek-cb — APPROVED** [review]

**Pull request merged**